motto/thread_agents/thread_agents.py

```diff
@@ -0,0 +1,84 @@
+from hyperon import *
+from motto.agents import DialogAgent
+from motto import get_sentence_from_stream_response
+from hyperon.ext import register_atoms
+from hyperon.exts.agents import BaseListeningAgent
+
+class ListeningAgent(DialogAgent, BaseListeningAgent):
```

Comment:
I guess we can avoid multiple inheritance here. Isn't `DialogAgent` a descendant of `BaseListeningAgent`? If not (or will not after refactoring `BaseListeningAgent` in hyperon-experimental, it may make sense to consider inheriting the base agent class in metta-motto from it.